### ISSUE TITLE
App using arcore-ios-sdk fails to deploy with xcode 15.3

### ISSUE THREAD

**Issue by jj0hns0n** [google-ar/arcore-ios-sdk#98]
We are seeing a similar issue to this https://github.com/firebase/firebase-ios-sdk/issues/12441 with an app that depends on arcore-ios-sdk using Xcode 15.3. 

Reverting to 15.2 works for now locally and on xcode cloud.

Locally, I see this error 

![Screenshot 2024-03-07 at 9 24 36 AM](https://github.com/google-ar/arcore-ios-sdk/assets/506078/3a08ccbe-8987-4b8a-aa27-765c7477e612)

```
Unable to Install “Vision Dev”
Domain: IXUserPresentableErrorDomain
Code: 1
Recovery Suggestion: Failed to load Info.plist from bundle at path /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework; Extra info about "/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework/Info.plist": Couldn't stat /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework/Info.plist: No such file or directory
User Info: {
    DVTErrorCreationDateKey = "2024-03-07 17:24:32 +0000";
    IDERunOperationFailingWorker = IDEInstallCoreDeviceWorker;
}
--
Unable to Install “Vision Dev”
Domain: IXUserPresentableErrorDomain
Code: 1
Recovery Suggestion: Failed to load Info.plist from bundle at path /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework; Extra info about "/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework/Info.plist": Couldn't stat /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework/Info.plist: No such file or directory
User Info: {
    IDERunOperationFailingWorker = IDEInstallCoreDeviceWorker;
}
--
Failed to install the app on the device.
Domain: com.apple.dt.CoreDeviceError
Code: 3002
User Info: {
    NSURL = "file:///Users/jjohnson/Library/Developer/Xcode/DerivedData/sspvision-aokmuwzcjjlfvagoiwmqixwxdxya/Build/Products/Debug-iphoneos/Vision%20Dev.app/";
}
--
Unable to Install “Vision Dev”
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Please try again later.
Recovery Suggestion: Failed to load Info.plist from bundle at path /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework; Extra info about "/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework/Info.plist": Couldn't stat /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework/Info.plist: No such file or directory
--
Failed to load Info.plist from bundle at path /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework; Extra info about "/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework/Info.plist": Couldn't stat /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.14AZCB/extracted/Vision Dev.app/Frameworks/ARCoreGARSession.framework/Info.plist: No such file or directory
Domain: MIInstallerErrorDomain
Code: 35
User Info: {
    FunctionName = "-[MIBundle _validateWithError:]";
    LegacyErrorString = PackageInspectionFailed;
    SourceFileLine = 64;
}
--

Event Metadata: com.apple.dt.IDERunOperationWorkerFinished : {
    "device_isCoreDevice" = 1;
    "device_model" = "iPhone13,3";
    "device_osBuild" = "17.3.1 (21D61)";
    "device_platform" = "com.apple.platform.iphoneos";
    "dvt_coredevice_version" = "355.24";
    "dvt_mobiledevice_version" = "1643.100.58";
    "launchSession_schemeCommand" = Run;
    "launchSession_state" = 1;
    "launchSession_targetArch" = arm64;
    "operation_duration_ms" = 1089;
    "operation_errorCode" = 1;
    "operation_errorDomain" = IXUserPresentableErrorDomain;
    "operation_errorWorker" = IDEInstallCoreDeviceWorker;
    "operation_name" = IDERunOperationWorkerGroup;
    "param_debugger_attachToExtensions" = 0;
    "param_debugger_attachToXPC" = 1;
    "param_debugger_type" = 3;
    "param_destination_isProxy" = 0;
    "param_destination_platform" = "com.apple.platform.iphoneos";
    "param_diag_MainThreadChecker_stopOnIssue" = 0;
    "param_diag_MallocStackLogging_enableDuringAttach" = 0;
    "param_diag_MallocStackLogging_enableForXPC" = 1;
    "param_diag_allowLocationSimulation" = 1;
    "param_diag_checker_tpc_enable" = 1;
    "param_diag_gpu_frameCapture_enable" = 0;
    "param_diag_gpu_shaderValidation_enable" = 0;
    "param_diag_gpu_validation_enable" = 1;
    "param_diag_memoryGraphOnResourceException" = 0;
    "param_diag_queueDebugging_enable" = 1;
    "param_diag_runtimeProfile_generate" = 0;
    "param_diag_sanitizer_asan_enable" = 0;
    "param_diag_sanitizer_tsan_enable" = 0;
    "param_diag_sanitizer_tsan_stopOnIssue" = 0;
    "param_diag_sanitizer_ubsan_stopOnIssue" = 0;
    "param_diag_showNonLocalizedStrings" = 0;
    "param_diag_viewDebugging_enabled" = 1;
    "param_diag_viewDebugging_insertDylibOnLaunch" = 1;
    "param_install_style" = 0;
    "param_launcher_UID" = 2;
    "param_launcher_allowDeviceSensorReplayData" = 0;
    "param_launcher_kind" = 0;
    "param_launcher_style" = 99;
    "param_launcher_substyle" = 8192;
    "param_runnable_appExtensionHostRunMode" = 0;
    "param_runnable_productType" = "com.apple.product-type.application";
    "param_structuredConsoleMode" = 1;
    "param_testing_launchedForTesting" = 0;
    "param_testing_suppressSimulatorApp" = 0;
    "param_testing_usingCLI" = 0;
    "sdk_canonicalName" = "iphoneos17.4";
    "sdk_osVersion" = "17.4";
    "sdk_variant" = iphoneos;
}
--


System Information

macOS Version 14.1.1 (Build 23B81)
Xcode 15.3 (22618) (Build 15E204a)
Timestamp: 2024-03-07T09:24:32-08:00
```

On App Store Connect the error looks like this.

<img width="893" alt="Screenshot 2024-03-07 at 2 50 48 PM" src="https://github.com/google-ar/arcore-ios-sdk/assets/506078/f52ceb30-2ddc-46a9-a29f-af232acac1fe">

**Comment by ertembiyik** [comment]
same issue

**Comment by VadimYakovliev** [comment]
same issue

**Comment by binhao94** [comment]
Experiencing the same issue too...

**Comment by AbsilDylan** [comment]
Same here

**Comment by GillDreamer** [comment]
Same issue

**Comment by 15kingben** [comment]
Sorry for the delay in fixing this issue, we are working on a fix for Xcode 15.3 support. A workaround is to build with Cocoapods on Xcode 15.3 or use Xcode 15.2

**Comment by arthurschiller** [comment]
Any estimates on a release? Thank you! 🙏

**Comment by 15kingben** [comment]
Unfortunately it will not be in the upcoming release, it will be in the one after

**Comment by arthurschiller** [comment]
> Unfortunately it will not be in the upcoming release, it will be in the one after

Any ideas when that would be? This drastically complicates my workflow since my app uses features that require iOS 17.2 and visionOS 1.2 (requiring Xcode 15.3) so now being forced to build with Xcode 15.2 to support ARCore is not ideal.

**Comment by 15kingben** [comment]
That release will be in late June. Note that ARCore is not supported on the Vision Pro

**Comment by VadimYakovliev** [comment]
Same issue with xcode 15.4

**Comment by vindicatesociety** [comment]
Did you try setting ENABLE_USER_SCRIPT_SANDBOXING to NO in the Build Settings?

**Comment by jrullman** [comment]
This issue is fixed with ARCore 1.44, which was released about a week ago. You need the newest ARCore, not Xcode.

(P.S. ENABLE_USER_SCRIPT_SANDBOXING=NO is necessary for Cocoapods, while this issue pertains to Swift Package Manager.)
